### ISSUE TITLE
Accessibility focus should not change after data refresh

### ISSUE THREAD

**Issue by geosaaga** [fmidev/weather-app#1188]
We should keep accessibility focus unchanged after data refresh. There are some ways to fix or at least improve current logic.

1. We should not hide actual UI during data refresh like for example `NextHourForecastPanelWithWeatherBackground` does. Loading indicator could be more minimalistic and included in the UI or over top of the current UI (Toast).
2. We should announce data refresh for screen reader so that if focus is lost anyway user can understand why it happened. `AccessibilityInfo.announceForAccessibility()` can be used for this.
3. We can try to restore focus after data refresh.
4. We should not update state and make UI refresh if data remains same. We can check is forecast `modtime` same in state and in new data.
